opslevel/resources_opslevel_team_tag.go

```diff
@@ -0,0 +1,157 @@
+package opslevel
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/opslevel/opslevel-go/v2022"
+)
+
+func resourceTeamTag() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manages a team tag",
+		Create:      wrap(resourceTeamTagCreate),
+		Read:        wrap(resourceTeamTagRead),
+		Update:      wrap(resourceTeamTagUpdate),
+		Delete:      wrap(resourceTeamTagDelete),
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"last_updated": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"team": {
+				Type:        schema.TypeString,
+				Description: "The id of the team that this will be added to.",
+				ForceNew:    true,
+				Optional:    true,
+			},
+			"team_alias": {
+				Type:        schema.TypeString,
+				Description: "The alias of the team that this will be added to.",
+				ForceNew:    true,
+				Optional:    true,
+			},
+			"key": {
+				Type:         schema.TypeString,
+				Description:  "The tag's key.",
+				ForceNew:     false,
+				Required:     true,
+				ValidateFunc: validation.StringMatch(TagKeyRegex, TagKeyErrorMsg),
+			},
+			"value": {
+				Type:        schema.TypeString,
+				Description: "The tag's value.",
+				ForceNew:    false,
+				Required:    true,
+			},
+		},
+	}
+}
+
+func resourceTeamTagCreate(d *schema.ResourceData, client *opslevel.Client) error {
+	team, err := findTeam("team_alias", "team", d, client)
+	if err != nil {
+		return err
+	}
+
+	input := opslevel.TagCreateInput{
+		Id: team.Id,
+
```

Comment:
We need to also specific this TagCreate is for "type" of "team" (is/was this implemented in opslevel-go too?)

Reply:
I was a bit confused how this was working for me considering I wasn't specifying the type, but I just figured out why (and I wonder if we did this with service tags intentionally) 

With our tag create mutation we let users specify the resource by alias or id. However specifying the type is only required when we are using the alias to find the resource. Additionally, even if a user in terraform uses the alias to specify the resource, you can see a few lines above this comment that we're finding the team using the alias and then using the id in TagCreateInput (which explains why we don't need type)

Reply:
🤯 - nice job!  Also thats good to know though we should probably document that in the GraphQL public API.